lib/new_relic/agent/transaction.rb

```diff
@@ -236,6 +237,7 @@ def initialize(category, options)
 
         @exceptions = {}
         @metrics = TransactionMetrics.new
+        @logs = PrioritySampledBuffer.new(NewRelic::Agent.instance.log_event_aggregator.capacity)
```

Comment:
Slightly subtle but we cap out each transaction at the per-harvest limit since it can't reasonably keep more than that anyway.